File: Easy/Python/1566. Detect Pattern of Length M Repeated K or More Times.py
class Solution(object):
    def containsPattern(self, arr, m, k):
        count = 0

        for i in range(len(arr) - m):
            count = count + 1 if arr[i] == arr[i + m] else 0
            if count == m * (k - 1):
                return True

        return False
        

File: Easy/JavaScript/1566. Detect Pattern of Length M Repeated K or More Times.js
var containsPattern = function(arr, m, k) {
    let count = 0;

    for(let i = 0; i < arr.length - m; i++){
        count = (arr[i] == arr[i + m] ? count + 1 : 0);
        if(count == m * (k - 1))
            return true;
    }

    return false;
};
